### PR TITLE
Added initial group POST tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
@paglias I'm unsure if this is the correct way to send updates to a separate branch than develop.

I have added the initial group POST tests. Everything passes when I comment out these:

```
firebase.updateGroupData(savedGroup);
      firebase.addUserToGroup(savedGroup._id, user._id);
```

I get an `unexpected error occured` during `gulp:test:api-v3:integration`. I was wondering if you new anything about the issue? If not, I will be reviewing it when I return tonight.

### CODE REVIEW

File: test/api/v3/integration/groups/POST-groups.test.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+  before(() => {
+     return generateUser().then((generatedUser) => {
+       user = generatedUser;
+       api = requester(user);
+     });
+   });
+
+   context('Guilds', () => {
+    it('user with insufficient funds from creating a guild', () => {
```
Comment (crookedneighbor) [inline test/api/v3/integration/groups/POST-groups.test.js]: Is this supposed to be `it('returns error when user with...` ?

File: test/api/v3/integration/groups/POST-groups.test.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+             type: groupType
+           });
+         })
+         .then((result) => {
+           expect(result._id).to.exist;
+           expect(result.name).to.equal(groupName);
+           expect(result.type).to.equal(groupType);
+         })
+         .then(() => {
+           api = requester(tmpUser);
```
Comment (crookedneighbor) [inline test/api/v3/integration/groups/POST-groups.test.js]: Why is this here?

**Comment by crookedneighbor** [comment]
This is the correct way to PR into another branch :+1:

File: test/api/v3/integration/groups/POST-groups.test.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        })
+        .then(() => {
+          return expect(api.post('/groups')).to.eventually.be.rejected.and.eql({
+            code: 401,
+            error: 'NotAuthorized',
+            message: t('messageGroupAlreadyInParty'),
+          });
+        })
+        .then(() => {
+          api = requester(tmpUser);
```
Comment (paglias) [inline test/api/v3/integration/groups/POST-groups.test.js]: same here, why do you change the requester user?

**Comment by paglias** [comment]
Except for the comments this looks good, I'll merge it and fix the errors when the comments are resolved

File: test/api/v3/integration/groups/POST-groups.test.js (hunk 3 above)
Comment (TheHollidayInn) [inline test/api/v3/integration/groups/POST-groups.test.js]: This is supposed to restore the user. I made a mistake. I was supposed to pass user. I changed the user, so this test would not depend on the previous test. As in, I greater a new user who tried to created two parties.

**Comment by TheHollidayInn** [comment]
Okay, I'll post the updates tonight. Did you know anything about the Firebase issues?

File: test/api/v3/integration/groups/POST-groups.test.js (hunk 3 above)
Comment (crookedneighbor) [inline test/api/v3/integration/groups/POST-groups.test.js]: You should just change the initial before to a beforeEach so a new user gets created every time.

**Comment by TheHollidayInn** [comment]
Alright, I made updates. If I shouldn't be changing the user, let me know. I think changing the user definitely make sense for the two part test though (or I could leave the party and create a new one)

**Comment by paglias** [comment]
I have not been able to test the firebase issue, will try tomorrow. Anyway
don't worry about it, it's probably an error I made
Il 19/dic/2015 01:42, "Keith Holliday" notifications@github.com ha
scritto:

**Comment by paglias** [comment]
@crookedneighbor this looks good to me so i'm merging it, let me know if you think anything has to be changed

**Comment by paglias** [comment]
@TheHollidayInn I'm trying to figure out why the tests are failing, one thing: you should pay attention to the indentation you use, we use 2 spaces for every level of indentation. We still don't have eslint for tests but you should try to it. Same thing for quotes: we use single quotes.

**Comment by TheHollidayInn** [comment]
Gotcha. I think I accidentally messed up the indention on these files, I usually do 2 spaces. I'll make sure to have the quotes done.

**Comment by paglias** [comment]
I've fixed all the tests, the build should be passing now

2015-12-19 18:21 GMT+01:00 Keith Holliday notifications@github.com:

> Gotcha. I think I accidentally messed up the indention on these files, I
> usually do 2 spaces. I'll make sure to have the quotes done.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/HabitRPG/habitrpg/pull/6380#issuecomment-166005438.

## 

Matteo Pagliazzi - paglias.net

**Comment by paglias** [comment]
@TheHollidayInn to better catch errors in tests you could do the following:
1. Open 2 terminal windows and paste in both
   `export API_VERSION="v3"; export PORT=3004; export NODE_DB_URI="mongodb://localhost/habitrpgtest";`
2. In one of the terminals type:
   `gulp nodemon`
3. In the other one you can run tests with `mocha test/api/v3/path` so for example `mocha test/api/v3/groups`
4. Now you can see any error happening on the server in the first terminal window

credits to @crookedneighbor for the very helpful trick :)

**Comment by TheHollidayInn** [comment]
This is great! I was wondering how to view this information. Thanks, @crookedneighbor !

**Comment by paglias** [comment]
@TheHollidayInn regarding my last message: until we fix a bug in the test suite it will fail for any test that directly edits the database (like setting user.balance on creation)
